resources/sdk/clisdkclient/templates/restclient.mustache

```diff
@@ -0,0 +1,192 @@
+package restclient
+
+import (
+	"net/http"
+
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"gc/config"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/tidwall/pretty"
+)
+
+var (
+	Client HTTPClient
+)
+
+// HTTPClient interface
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type HttpStatusError struct {
+	Verb       string
+	Path       string
+	Body       string
```

Comment:
@carnellj I added the Body to this struct so we can see the reason for the error